save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/TopBar.kt

```diff
@@ -54,6 +54,145 @@ external interface TopBarProps : PropsWithChildren {
     var userInfo: UserInfo?
 }
 
+/**
+ * The class for analyzing url address and creating links in topBar
+ */
+class TopBarUrl(href: String) {
+    /**
+     * currentPath is the link that we put in buttons
+     */
+    var currentPath = "#"
+    private var circumstance: SituationUrlClassification = SituationUrlClassification.KEYWORD_PROCESS
+
+    init {
+        circumstance = SituationUrlClassification.findExclude(href)
+    }
+
+    /**
+     * The function is called to specify the link address in the button before creating the button itself
+     *
+     * @param pathPart is an appended suffix to an already existing [currentPath]
+     */
+    fun changeUrlBeforeButton(pathPart: String) {
+        currentPath = SituationUrlClassification.fixCurrentPathBefore(circumstance, pathPart, currentPath)
+    }
+
+    /**
+     * The function is called to specify the link address in the button after creating the button itself
+     *
+     * @param pathPart is an appended suffix to an already existing [currentPath]
+     */
+    fun changeUrlAfterButton(pathPart: String) {
+        currentPath = SituationUrlClassification.fixCurrentPathAfter(circumstance, pathPart, currentPath)
+        circumstance = SituationUrlClassification.fixExcludeAfter(circumstance, "\\d+".toRegex().matches(pathPart))
+    }
+
+    /**
+     * The function returns a flag whether to create this button or not
+     *
+     * @param index is index of this segment in url address
+     */
+    fun isCreateButton(index: Int) = SituationUrlClassification.isCreateButton(circumstance, index)
+
+    private enum class SituationUrlClassification {
+        ARCHIVE,  // situation with the processing of the "archive" in the url address - need for tabs in AwesomeBenchmarksView
+        DETAILS,  // situation with the processing of the "details" in the url address - need for deleted multi-segment urls, starting with the word "details"
+        EXECUTION,  // situation with the processing of the "execution" in the url address - need for redirect to the page with the executions history
+        KEYWORD_NOT_PROCESS,  // the button with this url segment is not created
+        KEYWORD_PROCESS,  // the button with this url segment is created without changes
+        KEYWORD_PROCESS_LAST_SEGMENTS,  // a button is created if this segment is one of the last
+        PROJECT_OR_ORGANIZATION,  // situation with the processing of the "archive" in the url address - need for tabs in OrganizationView and ProjectView,
+        ;
+
+        companion object {
```

Comment:
Why are all the methods inside `companion object` of enum?

Reply:
Yes, all work with Enum elements lies in Enum class

Reply:
To my mind all the analysis takes place in `TopBarUrl` and it might be more clear to make all the methods connected with url analysis be part of `TopBarUrl`

Reply:
Ok, done